materials_io/utils/__init__.py

```diff
@@ -0,0 +1,178 @@
+from typing import Dict, Union, Tuple, Any, Callable, Optional, List
+from typing_extensions import TypedDict
+
+
+def get_nested_dict_value_by_path(nest_dict: Dict,
+                                  path: Union[Tuple, str],
+                                  cast: Optional[Callable] = None) -> Any:
+    """
+    Get the value from within a nested dictionary structure by traversing into
+    the dictionary as deep as that path found and returning that value
+
+    Parameters
+    ----------
```

Comment:
Minor thing: I've been using the Google-style docstrings in other parts. I wouldn't recommend switching these, but would you be OK with Google-style going forward.

Reply:
Sorry, habit to use NumpyDoc style. I can switch them to Google-style so it's consistent